lib/jobs/obm-control.js

```diff
@@ -64,18 +62,33 @@ function obmControlJobFactory(
             assert.ok(node, "Node should exist to run OBM command");
             assert.ok(node.obmSettings, "Node should have OBM settings to run OBM command");
 
-            var settings = _.find(node.obmSettings, function(setting) {
+            self.settings = _.find(node.obmSettings, function(setting) {
                 return setting ? setting.service === self.obmServiceName : false;
             });
 
-            assert.ok(settings, "Node should have OBM settings for service: " +
-                self.obmServiceName);
+            if (self.obmServiceName) {
+                // throw error if the exact obm service doesn't exist
+                assert.ok(self.settings, "Node should have OBM settings for service: " +
+                    self.obmServiceName);
+            } else {
+                //find the default obm service
+                if(node.obmSettings.length === 0){
+                    assert.fail(true,true, "No OBM service assigned to this node");
+                }else if(node.obmSettings.length === 1) {
+                    self.settings = node.obmSettings[0];
+                    logger.info("Defaulting to obmServiceName "+ self.settings.service +
+                    " for node because only one OBM service present.");
+                }else if(node.obmSettings.length > 1){
```

Comment:
ok, so I refactored this to use `else`, and in addition, folded in the use case for `node.obmSettings.length === 0` because I was finding I could not hit that branch of code with tests, due to the assert found on line 63, it seemed to always trigger when the property either does not exist or is an empty array.